com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -1994,8 +1996,29 @@ private void OnClientLoadedScene(uint sceneEventId, Scene scene)
         /// synchronized.
         /// </summary>
         /// <param name="clientId">newly joined client identifier</param>
-        internal void SynchronizeNetworkObjects(ulong clientId)
+        internal void SynchronizeNetworkObjects(ulong clientId, bool synchronizingService = false)
         {
+            // If we are connected to a live service hosted session and we are not doing the initial synchronization for the service...
+            if (NetworkManager.CMBServiceConnection && !synchronizingService)
+            {
+                // then as long as this is a newly connecting client add it to the connecting client queue.
+                // Otherwise, if this is not a newly connecting client (i.e. it is already in the queue), then go ahead and synchronize
+                // that client.
+                if (!ClientConnectionQueue.Contains(clientId))
+                {
+                    ClientConnectionQueue.Add(clientId);
+                    // If we are already synchronizing one or more clients, then add this client to the queue and exit.
```

Comment:
This comment is a little confusing as it's after the `Add`. Potentially this is clearer?
```suggestion
                    // If we are already synchronizing one or more clients, exit early. This client will be synchronized later.
```